frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -294,18 +364,63 @@ pub mod pallet {
 						pool_id_and_state.state == State::Normal,
 						Error::<T>::TransferringInProgress
 					);
-					Pools::<T>::mutate(asset_id, |_| PoolState { pool_id, state: State::Normal });
+					Pools::<T>::mutate(asset_id, |pool| {
+						*pool = Some(PoolState { pool_id, state: State::Transferring });
+					});
+					Self::deposit_event(Event::OccuredFundsTransferring {
+						old_pool_id: pool_id_and_state.pool_id,
+						new_pool_id: pool_id,
+					});
+					Self::transferring_funds_from_old_pool_to_new(
+						asset_id,
+						pool_id_and_state.pool_id,
+						pool_id,
+					)?;
 				},
 				Err(_) => Pools::<T>::insert(asset_id, PoolState { pool_id, state: State::Normal }),
 			}
 			Ok(())
 		}
+		/// Called when an existing asset_id-pool_id pair is replaced by new pool_id.
+		/// Occurs a withdrawal of all funds from the old pool to the Vault, and then from the Vault
+		/// to the new pool.
+		#[transactional]
+		fn transferring_funds_from_old_pool_to_new(
+			asset_id: T::AssetId,
+			old_pool_id: T::PoolId,
+			new_pool_id: T::PoolId,
+		) -> DispatchResult {
+			AssociatedVaults::<T>::try_mutate(|vaults| {
+				// TODO(belousm): Ask Kevin about uniquest of vaults with underlying asset
```

Comment:
// TODO(belousm): Ask Kevin about uniquest of vaults with underlying asset

There will be a one to one ratio of whitelisted assets to cubic vaults created by Instrumental.

Because of that I believe we should make a storage object that can store the reference from asset_ids to associated vaults, that way you don't have to cycle through all vaults to find the one that has the correct asset_id.

A downside to this approach is that we have to make sure we update this storage object when we change which strategies are associated with a vault.